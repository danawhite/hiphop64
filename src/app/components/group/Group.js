import React from 'react';

import GroupCheckbox from '../group-checkbox/GroupCheckbox';

export default class Group extends React.Component {
    constructor(props){
        super(props);
        //console.log(this);

        this.style = {
            container: {
                backgroundColor: '#fafafa',
                display: 'block',
                border: '1px solid green',
                width: 180,
                height: 25,
                verticalAlign: 'center',
                cursor: 'pointer'
            },
            nodes: {
                display: 'inline',
                paddingLeft: 5,
                paddingRight: 10
            },
            checkbox: {
                alignSelf: 'flex-end'
            }
        };

        this.setSelectedGroup = this.setSelectedGroup.bind(this);
        this.handleClick = this.handleClick.bind(this);
    }

    setSelectedGroup(props) {
        console.log(props);
    }

    handleClick(props) {
        this.props.onSelected(this.props);
    }

    render() {
        return (
            <div style={this.style.container} onClick={this.handleClick}>
                <p style={this.style.nodes}>{this.props.seed}</p>
                <p style={this.style.nodes}>{this.props.name}</p>
                <input type="checkbox" style={this.style.checkbox}/>
            </div>
        )
    }
}

Group.propTypes = {
    //seed: React.PropTypes
    //    .oneOfType([
    //        React.propTypes.string,
    //        React.propTypes.number
    //    ]),
    //name: React.PropTypes.string
};

Group.defaultProps = {};
