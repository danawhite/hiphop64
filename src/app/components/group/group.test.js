import React from 'react';
import Group from './group';
import renderer from 'react-test-renderer';

describe('Group', () => {

    it('renders correctly', () => {
        const tree = renderer.create(
            <Group name="Onyx" seed="14"/>
        ).toJSON();
        expect(tree).toMatchSnapshot()
    })
});

//    it('should include a prop seed of type string', () => {
//        expect(group).not.toBeDefined();
//    });
//
//    it('should send GROUP_SELECTED action when clicked', () => {
//        TestUtils.Simulate.click(
//            TestUtils.findRenderedDOMComponentWithTag(group, 'div')
//        );
//    })

