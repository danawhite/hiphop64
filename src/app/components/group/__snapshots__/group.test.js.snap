exports[`Group renders correctly 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#ffdd00",
      "border": "1px solid green",
      "cursor": "pointer",
      "display": "block",
      "height": 25,
      "verticalAlign": "center",
      "width": 180,
    }
  }>
  <p
    style={
      Object {
        "display": "inline",
        "paddingLeft": 5,
        "paddingRight": 10,
      }
    }>
    14
  </p>
  <p
    style={
      Object {
        "display": "inline",
        "paddingLeft": 5,
        "paddingRight": 10,
      }
    }>
    Onyx
  </p>
  <input
    style={
      Object {
        "alignSelf": "flex-end",
      }
    }
    type="checkbox" />
</div>
`;
